.github/workflows/api-binary-compatibility.yml

```diff
@@ -27,6 +27,9 @@ on:
       - 'apache-iceberg-**'
   pull_request:
     paths:
+      - '.github/**'
```

Comment:
seems unrelated to the scope of the PR

Reply:
it's also extracted as https://github.com/apache/iceberg/pull/10485, see rationale there.

this PR changes build system (gradle version) and it would not be smart to assume workflows continue to work without running them

Of course, this doesn't prevent all problems -- revapi silently not working is a type of problem this doesn't prevent -- but at least prevents some and in that sense it is related to the scope of this PR.